addons/hr_work_entry_contract/static/src/js/work_entries_controller_mixin_owl.js

```diff
@@ -0,0 +1,46 @@
+/** @odoo-module **/
+
+import { serializeDate } from "@web/core/l10n/dates";
+import { useService } from "@web/core/utils/hooks";
+
+export function useWorkEntry() {
+    const orm = useService("orm");
+    const action = useService("action");
+
+    function getEmployeeIds(model) {
+        const getEmployeeIdFromRecord = (record) => {
+            return record.rawRecord.employee_id[0];
+        }
+
+        const groups = Object.values(model.records).reduce((groups, record) => {
+            groups.add(getEmployeeIdFromRecord(record));
+            return groups;
+        }, new Set());
+        return Array.from(groups);
+    }
```

Comment:
Reduce is nice but most of the time it's overkill and does not convey the code's intention explicitely enough (here though the method name makes it clear enough).
NOTE: this function should be passed to the hook as well.
```suggestion
    function getEmployeeIds(model) {
        return [...new Set(Object.values(model.records).map(rec => rec.rawRecord.employee_id[0]))];
    }
```